server/src/utils/common.ts

```diff
@@ -67,6 +125,115 @@ let polisTypes: PolisTypes = {
 polisTypes.reactionValues = _.values(polisTypes.reactions);
 polisTypes.starValues = _.values(polisTypes.staractions);
 
-export { strToHex, hexToStr, polisTypes };
+function isConversationOwner(
+  zid: any,
+  uid?: any,
+  callback?: {
+    (err: any): void;
+    (err: any): void;
+    (err: any): void;
+    (err: any, foo: any): void;
+    (err: any, foo: any): void;
+    (arg0: any): void;
+  }
+) {
+  pgQuery_readOnly(
+    "SELECT * FROM conversations WHERE zid = ($1) AND owner = ($2);",
+    [zid, uid],
+    function (err: number, docs: { rows: string | any[] }) {
+      if (!docs || !docs.rows || docs.rows.length === 0) {
+        err = err || 1;
+      }
+      callback?.(err);
+    }
+  );
+}
+
+function isModerator(zid: any, uid?: any) {
+  if (isPolisDev(uid)) {
+    return Promise.resolve(true);
+  }
+  return pgQueryP_readOnly(
+    "select count(*) from conversations where owner in (select uid from users where site_id = (select site_id from users where uid = ($2))) and zid = ($1);",
+    [zid, uid]
+    //     Argument of type '(rows: { count: number; }[]) => boolean' is not assignable to parameter of type '(value: unknown) => boolean | PromiseLike<boolean>'.
+    // Types of parameters 'rows' and 'value' are incompatible.
+    //     Type 'unknown' is not assignable to type '{ count: number; }[]'.ts(2345)
+    // @ts-ignore
+  ).then(function (rows: { count: number }[]) {
+    return rows[0].count >= 1;
+  });
+}
+
+function doAddDataExportTask(
+  math_env: string | undefined,
+  email: string,
+  zid: number,
+  atDate: number,
+  format: string,
+  task_bucket: number
+) {
+  return pgQueryP(
+    "insert into worker_tasks (math_env, task_data, task_type, task_bucket) values ($1, $2, 'generate_export_data', $3);",
+    [
+      math_env,
+      {
+        email: email,
+        zid: zid,
+        "at-date": atDate,
+        format: format,
+      },
+      task_bucket, // TODO hash the params to get a consistent number?
+    ]
+  );
+}
+
+function isOwner(zid: any, uid: string) {
+  return Conversation.getConversationInfo(zid).then(function (info: any) {
+    return info.owner === uid;
+  });
+}
 
-export default { strToHex, hexToStr, polisTypes };
+const escapeLiteral = pg.Client.prototype.escapeLiteral;
+
+function isDuplicateKey(err: {
+  code: string | number;
+  sqlState: string | number;
+  messagePrimary: string | string[];
+}) {
+  let isdup =
+    err.code === 23505 ||
+    err.code === "23505" ||
+    err.sqlState === 23505 ||
+    err.sqlState === "23505" ||
+    (err.messagePrimary && err.messagePrimary.includes("duplicate key value"));
+  return isdup;
+}
+
+export {
```

Comment:
Exporting everything twice is an anti-pattern. Determine which type of export to use for this module (default or non-default) and remove the other one.